routing/router.go

```diff
@@ -1595,14 +1595,50 @@ type LightningPayment struct {
 // within the network to reach the destination. Additionally, the payment
 // preimage will also be returned.
 func (r *ChannelRouter) SendPayment(payment *LightningPayment) ([32]byte, *route.Route, error) {
+	paySession, err := r.preparePayment(payment)
+	if err != nil {
+		return [32]byte{}, nil, err
+	}
+
+	// Since this is the first time this payment is being made, we pass nil
+	// for the existing attempt.
+	return r.sendPayment(nil, payment, paySession)
+}
+
+// SendPaymentAsync is the non-blocking version of SendPayment. The payment
+// result needs to be retrieved via the control tower.
+func (r *ChannelRouter) SendPaymentAsync(payment *LightningPayment) error {
+	paySession, err := r.preparePayment(payment)
```

Comment:
I think this can just be:
```go
go func() {
    if _, _, err := r.SendPayment(payment); err != nil {
			log.Errorf("Payment with hash %v failed: %v",
				payment.PaymentHash, err)
    }
}

return nil
```

Reply:
If so, then we also wouldn't need to extract logic into `preparePayment`.

Reply:
We don't want to return before the payment is registered with the control tower. When the rpc call returns, the caller should have the guarantee that their payment is started and will be resumed on restart.

Reply:
Comment above isn't applicable anymore because `SendPayment` is blocking now.